charmrepo/legacy_test.go

```diff
@@ -371,6 +371,37 @@ func (s *legacyCharmStoreSuite) TestCharmURL(c *gc.C) {
 	}
 }
 
+var legacyInferRepositoryTests = []struct {
+	url  string
+	path string
+}{
+	{"cs:precise/wordpress", ""},
+	{"local:oneiric/wordpress", "/some/path"},
+}
+
+func (s *legacyCharmStoreSuite) TestInferRepository(c *gc.C) {
+	for i, t := range legacyInferRepositoryTests {
+		c.Logf("test %d", i)
+		ref, err := charm.ParseReference(t.url)
+		c.Assert(err, gc.IsNil)
+		repo, err := charmrepo.LegacyInferRepository(ref, "/some/path")
+		c.Assert(err, gc.IsNil)
+		switch repo := repo.(type) {
+		case *charmrepo.LocalRepository:
+			c.Assert(repo.Path, gc.Equals, t.path)
+		default:
+			c.Assert(repo, gc.Equals, charmrepo.LegacyStore)
+		}
+	}
+	ref, err := charm.ParseReference("local:whatever")
+	c.Assert(err, gc.IsNil)
+	_, err = charmrepo.LegacyInferRepository(ref, "")
+	c.Assert(err, gc.ErrorMatches, "path to local repository not specified")
+	ref.Schema = "foo"
+	_, err = charmrepo.LegacyInferRepository(ref, "")
+	c.Assert(err, gc.ErrorMatches, "unknown schema for charm reference.*")
+}
+
```

Comment:
What if "cs:something" is passed the path string, is the paths skipped and only CS part taken for inferring repository?